### PR TITLE
[docs] Update DebuggingTheCompiler.md

### PR DESCRIPTION
**Pull request by benrimmington**
Follow-up to: #31502 and #31630

### CODE REVIEW

**Comment by benrimmington** [comment]
Cc: @gottesmm

File: docs/DebuggingTheCompiler.md [hunk 1]
```diff
@@ -380,18 +380,18 @@ and check for the function name in the breakpoint condition:
 
 Sometimes you may want to know which optimization inserts, removes or moves a
 certain instruction. To find out, set a breakpoint in
-`ilist_traits<SILInstruction>:addNodeToList` or
-`ilist_traits<SILInstruction>:removeNodeFromList`, which are defined in
+`ilist_traits<SILInstruction>::addNodeToList` or
```
Comment (harlanhaskins) [inline docs/DebuggingTheCompiler.md:383]: These look like unintentional fallout of the `::` -> `:` replacement

**Review by gottesmm — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci smoke test and merge

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
⛵

**Pull request merged**